src/txn/txn_timestamp.c

```diff
@@ -914,46 +953,70 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     __wt_config_init(session, &cparser, cfg[1]);
     while ((ret = __wt_config_next(&cparser, &ckey, &cval)) == 0) {
         WT_ASSERT(session, ckey.str != NULL);
-        if (WT_STRING_MATCH("commit_timestamp", ckey.str, ckey.len)) {
+        if (WT_STRING_MATCH("commit_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "commit", &commit_ts, &cval));
-            set_ts = true;
-        } else if (WT_STRING_MATCH("durable_timestamp", ckey.str, ckey.len)) {
+        else if (WT_STRING_MATCH("durable_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "durable", &durable_ts, &cval));
-            set_ts = true;
-        } else if (WT_STRING_MATCH("prepare_timestamp", ckey.str, ckey.len)) {
+        else if (WT_STRING_MATCH("prepare_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "prepare", &prepare_ts, &cval));
-            set_ts = true;
-        } else if (WT_STRING_MATCH("read_timestamp", ckey.str, ckey.len)) {
+        else if (WT_STRING_MATCH("read_timestamp", ckey.str, ckey.len))
             WT_RET(__wt_txn_parse_timestamp(session, "read", &read_ts, &cval));
-            set_ts = true;
-        }
     }
     WT_RET_NOTFOUND_OK(ret);
 
-    /* Look for a commit timestamp. */
-    if (commit_ts != WT_TS_NONE)
-        WT_RET(__wt_txn_set_commit_timestamp(session, commit_ts));
-
-    /*
-     * Look for a durable timestamp. Durable timestamp should be set only after setting the commit
-     * timestamp.
-     */
-    if (durable_ts != WT_TS_NONE)
-        WT_RET(__wt_txn_set_durable_timestamp(session, durable_ts));
-    __wt_txn_publish_durable_timestamp(session);
+    return (__txn_set_timestamp_internal(session, commit_ts, durable_ts, prepare_ts, read_ts));
+}
 
-    /* Look for a read timestamp. */
-    if (read_ts != WT_TS_NONE)
-        WT_RET(__wt_txn_set_read_timestamp(session, read_ts));
+/*
+ * __ts_txn_type_to_str --
+ *     Convert a WT_TS_TXN_TYPE to a printable string.
+ */
+static const char *
+__ts_txn_type_to_str(WT_TS_TXN_TYPE which)
+{
+    switch (which) {
+    case WT_TS_TXN_TYPE_COMMIT:
+        return "commit_timestamp";
+    case WT_TS_TXN_TYPE_DURABLE:
+        return "durable_timestamp";
+    case WT_TS_TXN_TYPE_PREPARE:
+        return "prepare_timestamp";
+    case WT_TS_TXN_TYPE_READ:
+        return "read_timestamp";
+    }
 
-    /* Look for a prepare timestamp. */
-    if (prepare_ts != WT_TS_NONE)
-        WT_RET(__wt_txn_set_prepare_timestamp(session, prepare_ts));
+    return "UNKNOWN";
```

Comment:
The other names include "timestamp", and the error message is written assuming that context comes from the string, maybe update this to be "UNKNOWN timestamp"?

Reply:
Good idea, thanks.

Reply:
Fixed in bc79889